contract/r/gnoswap/gov/staker/v1/delegation_test.gno

```diff
@@ -695,3 +696,239 @@ func TestDelegation_EdgeCases(t *testing.T) {
 		})
 	}
 }
+
+// TestDelegation_addToCollectedAmount tests the private addToCollectedAmount function
+func TestDelegation_addToCollectedAmount(t *testing.T) {
+	tests := []struct {
+		name             string
+		collectedAmount  int64
+		amount           int64
+		expectedResult   int64
+		expectedHasError bool
+	}{
+		{
+			name:             "add positive amount",
+			collectedAmount:  100,
+			amount:           50,
+			expectedResult:   150,
+			expectedHasError: false,
+		},
+		{
+			name:             "add zero amount",
+			collectedAmount:  100,
+			amount:           0,
+			expectedResult:   100,
+			expectedHasError: false,
+		},
+		{
+			name:             "add negative amount",
+			collectedAmount:  100,
+			amount:           -30,
+			expectedResult:   0,
+			expectedHasError: true, // Should not allow negative amounts
+		},
+		{
+			name:             "both zero",
+			collectedAmount:  0,
+			amount:           0,
+			expectedResult:   0,
+			expectedHasError: false,
+		},
+		{
+			name:             "large values",
+			collectedAmount:  9223372036854775000,
+			amount:           100,
+			expectedResult:   9223372036854775100,
+			expectedHasError: false,
+		},
+		{
+			name:             "overflow",
+			collectedAmount:  math.MaxInt64, // MaxInt64
+			amount:           1,
+			expectedResult:   0,
+			expectedHasError: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// when
+			result, err := addToCollectedAmount(tc.collectedAmount, tc.amount)
+
+			// then
+			if tc.expectedHasError {
+				uassert.NotNil(t, err)
+			} else {
+				uassert.Nil(t, err)
+				uassert.Equal(t, result, tc.expectedResult)
+			}
+		})
+	}
+}
+
+// TestDelegation_min tests the private min function
+func TestDelegation_min(t *testing.T) {
```

Comment:
The `min` function only compares integer values using compare operators internally, so this test ends up testing the language specification itself. I think, this appears to be an unnecessary test. how do you think?